presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestTablePartitioningSelect.java

```diff
@@ -92,31 +96,15 @@ public void testSelectPartitionedHiveTableDifferentFormats()
     {
         String tableNameInDatabase = tablesState.get(TABLE_NAME).getNameInDatabase();
 
-        String selectFromAllPartitionsSql = "SELECT * FROM " + tableNameInDatabase;
-        QueryResult allPartitionsQueryResult = query(selectFromAllPartitionsSql);
-        assertThat(allPartitionsQueryResult).containsOnly(row(42, 1), row(42, 2));
-        assertProcessedLinesCountEquals(selectFromAllPartitionsSql, allPartitionsQueryResult, 2);
-
         String selectFromOnePartitionsSql = "SELECT * FROM " + tableNameInDatabase + " WHERE part_col = 2";
         QueryResult onePartitionQueryResult = query(selectFromOnePartitionsSql);
         assertThat(onePartitionQueryResult).containsOnly(row(42, 2));
-        assertProcessedLinesCountEquals(selectFromOnePartitionsSql, onePartitionQueryResult, 1);
-    }
-
-    private static final long GET_PROCESSED_LINES_COUNT_RETRY_SLEEP = 500;
 
-    private void assertProcessedLinesCountEquals(String sqlStatement, QueryResult allPartitionsQueryResult, int expected)
-            throws SQLException
-    {
-        long processedLinesCountAllPartitions = getProcessedLinesCount(sqlStatement, allPartitionsQueryResult);
-        if (processedLinesCountAllPartitions != expected) {
-            try {
-                Thread.sleep(GET_PROCESSED_LINES_COUNT_RETRY_SLEEP);
-            }
-            catch (InterruptedException e) {
-            }
-            processedLinesCountAllPartitions = getProcessedLinesCount(sqlStatement, allPartitionsQueryResult);
+        try {
+            // This query should fail or return null values for invalid partition data
+            assertThat(query("SELECT * FROM " + tableNameInDatabase)).containsOnly(row(42, 2), row(null, 1));
+        }
+        catch (QueryExecutionException expectedDueToInvalidPartitionData) {
```

Comment:
each format has its own message, they are totally different. I can assert if exception message is matching any of them but I did not like this.